mirar/processors/astrometry/anet/anet_processor.py

```diff
@@ -132,36 +204,57 @@ def _apply_to_images(self, batch: ImageBatch) -> ImageBatch:
                 timeout=self.timeout,
                 use_sextractor=self.use_sextractor,
                 sextractor_path=sextractor_path,
-                sextractor_config_path=self.sextractor_config_path,
+                sextractor_config_path=sextractor_config_path,
                 search_radius_deg=self.search_radius_deg,
                 parity=self.parity,
                 x_image_key=self.x_image_key,
                 y_image_key=self.y_image_key,
                 sort_key_name=self.sort_key_name,
             )
 
+            if self.write_regions:
+                coords_file = anet_out_dir.joinpath(
+                    image[BASE_NAME_KEY].replace(".fits", ".axy")
+                )
+                regions_path = anet_out_dir.joinpath(image[BASE_NAME_KEY] + ".reg")
+                logger.debug(f"Loading coords from {coords_file}")
+                with fits.open(coords_file) as hdul:
+                    coords_table = Table(hdul[1].data)  # pylint: disable=no-member
+                    write_regions_file(
+                        regions_path=regions_path,
+                        x_coords=coords_table[self.x_image_key],
+                        y_coords=coords_table[self.y_image_key],
+                        system="image",
+                    )
+
             newname = anet_out_dir.joinpath(Path(str(temp_path).split("temp_")[1]))
             if not newname.exists():
                 raise AstrometryNetError(
                     f"AstrometryNet did not run successfully - no output "
                     f"file {newname} found."
                 )
-            solved = fits.open(newname)
-            hdr = solved[0].header  # pylint: disable=no-member
-
-            del hdr["HISTORY"]
 
-            fits.writeto(  # pylint: disable=no-member
-                newname,
-                fits.open(temp_files[0])[0].data,  # pylint: disable=no-member
-                hdr,
-                overwrite=True,
-            )  # pylint: disable=no-member
-            batch[i] = self.open_fits(newname)  # pylint: disable=no-member
+            with fits.open(newname) as hdul:  # pylint: disable=no-member
+                hdr = hdul[0].header
+                data = hdul[0].data  # pylint: disable=no-member
+                if "HISTORY" in hdr:
+                    del hdr["HISTORY"]
 
+            # fits.writeto(  # pylint: disable=no-member
```

Comment:
@saarahha I commented out this piece as it seems like it was redundant. Can you please take a look to confirm whether that's indeed the case?